darshan-util/pydarshan/darshan/lib/accum.py

```diff
@@ -0,0 +1,116 @@
+    # a given module -- this information was commonly reported
+    # in the old perl-based summary reports
+    darshan_derived_metrics = log_get_derived_metrics(log_path=log_path,
+                                                      mod_name=mod_name)
+    total_mib = darshan_derived_metrics.total_bytes / 2 ** 20
+    total_bw = darshan_derived_metrics.agg_perf_by_slowest
+    ret_str = f"I/O performance estimate (at the {mod_name} layer): transferred {total_mib:.1f} MiB at {total_bw:.2f} MiB/s"
+    return ret_str
+
+
+def log_file_count_summary_table(log_path: str,
+                                 module: str):
+    # the darshan_file_category enum is not really
+    # exposed in CFFI/Python layer, so we effectively
+    # re-export the content indices we need here
+    # so that we can properly index the C-level data
+    darshan_file_category = {"total opened":0,
+                             "read-only files":1,
+                             "write-only files":2,
+                             "read/write files":3}
+    df = pd.DataFrame.from_dict(darshan_file_category, orient="index")
+    df.rename(columns={0:"index"}, inplace=True)
+    df.index.rename('type', inplace=True)
+    df["number of files"] = np.zeros(4, dtype=int)
+    df["avg. size"] = np.zeros(4, dtype=str)
+    df["max size"] = np.zeros(4, dtype=str)
+
+    darshan_derived_metrics = log_get_derived_metrics(log_path=log_path,
+                                                      mod_name=module)
+    for cat_name, index in darshan_file_category.items():
+        cat_counters = darshan_derived_metrics.category_counters[index]
+        num_files = int(cat_counters.count)
+        if num_files == 0:
+            max_size = "0"
+            avg_size = "0"
+        else:
+            max_size, binary_units = humanize.naturalsize(cat_counters.max_offset_bytes + 1,
+                                                          binary=True).split()
+            # some rounding work to match the old Perl
+            # report table style...
+            max_size = math.ceil(float(max_size)) # type: ignore
+            if max_size == 0:
+                max_size = "0"
+            else:
+                max_size = f"{max_size} {binary_units}"
+            # NOTE: internal formula based on discussion with Phil Carns
+            avg_size, binary_units = humanize.naturalsize((cat_counters.total_max_offset_bytes + num_files) / num_files,
+                                                           binary=True).split()
+            avg_size = math.ceil(float(avg_size)) # type: ignore
+            if avg_size == 0:
+                avg_size = "0"
+            else:
+                avg_size = f"{avg_size} {binary_units}"
+        df.iloc[index] = [index, num_files, avg_size, max_size]
+    # we don't need the index column once we're done with
+    # the CFFI/C interaction
+    df.drop(columns="index", inplace=True)
+    ret = plot_common_access_table.DarshanReportTable(df,
```

Comment:
Could we generalize DarshanReportTable, so that it isn't tied to the `plot_common_access_table` module (here or maybe we can open an issue if it's involved)? I guess they are the only existing tables in the report, but a "common access table" is a specific table instance and it will be awkward to have to create all tables this way.

Reply:
I think you're just repeating my review comment from https://github.com/darshan-hpc/darshan/pull/618#issuecomment-1018889780

> DarshanReportTable should be in a general data structure-related module somewhere most likely, I'll leave that up to you to discuss with the team.

Reply:
I opened gh-892 for you

Reply:
okay, thanks, i'll resolve this for now and look to review/merge gh-892 after this is merged.